docs/discussion.md

```diff
@@ -0,0 +1,232 @@
+&\quad+(\mu^3-2\mu^2+\mu)\cdot (x_1-x_0)&\cdot\quad  &f'(0)\\
+&\quad+(\mu^3-\mu^2)\cdot (x_1-x_0)&\cdot\quad &f'(1)\\
+\end{align*}
+```
+
+Whereas we know the function values, $f(0)$ and $f(1)$, we can only estimate the slopes, $f'(0)$ and $f'(1)$, from the data, itself. Several methods for estimating these slopes have been proposed.[^2] A common assumption is that the slope at any grid point is equal to that of the line containing the previous and next points on the axis:
+
+&ensp; &ensp; $f'(0)=(f(1)-f(-1))/(x_1-x_{-1})$
+
+&ensp; &ensp; $f'(1)=(f(2)-f(0))/(x_2-x_0)$
+
+<p align="center">
+    <img width="70%" src="./figs/fig_slope_est.png"> 
+</p>
+
+We define:
+
+&ensp; &ensp; $c_{0}=2\mu^3-3\mu^2+1, \quad  c_{1}=-2\mu^3+3\mu^2$
+
+&ensp; &ensp; $d_{0}=\mu^3-2\mu^2+\mu, \quad d_{1}=\mu^3-\mu^2$
+
+&ensp; &ensp; $s_0=(x_1-x_0)/(x_1-x_{-1}), \quad s_1=(x_1-x_0)/(x_2-x_0)$
+
+then rewrite the polynomial:   
+
+```math
+\begin{align*}
+f(\mu)=
+&\quad\quad c_{0}&\cdot\quad &f(0)\\
+&\quad+c_{1}&\cdot\quad &f(1)\\
+&\quad+d_{0}\cdot s_0&\cdot\quad &(f(1)-f(-1))\\
+&\quad+d_{1}\cdot s_1&\cdot\quad &(f(2)-f(0))
+\end{align*}
+```
+
+Finally, identify the coefficients:
+
+```math
+\begin{align*}
+f(\mu)=&-d_{0}\cdot s_0&\cdot\quad &f(-1)\\
+&+(c_{0} -d_{1}\cdot s_1)&\cdot \quad &f(0)\\
+&+(c_{1}+d_{0}\cdot s_0)&\cdot\quad &f(1)\\
+&+d_{1}\cdot s_1&\cdot\quad &f(2)
+\end{align*}
+```
+
+Terms $c_{0}$ and $c_{1}$ are referred to as *interpolation_coefficients*.
+
+The terms $d_{0}\cdot s_0$ and  $d_{1}\cdot s_1$ are referred to as *cubic_slope_coefficients*.
+
+<p align="center">
+    <img width="70%" src="./figs/fig_1D_cubic.png"> 
+</p>
+
+The above formulation allows cubic interpolation using the four function values associated with the grid-point coordinates in the vicinity of the queried coordinate. Notice that our slope estimations must be revised for edge cases, i.e., when either $x_0$ or $x_1$ represent the smallest or largest values, respectively, of the provided grid-axis coordinates. These cases can be handled by assigning the slope $(f(1)-f(0))/(x_1-x_0)$ to either $f'(0)$ or $f'(1)$, if either $x_{-1}$ or $x_2$, respectively, are outside the valid interpolation range.
+
+An equivalent method is used by **btwxt** to enforce this: If $x_0$ is the minimum coordinate, the assigments
+
+&ensp; &ensp; $s_0\gets 1$, $f(-1)\gets f(0)$
+
+are applied. If $x_1$ is the maximum coordinate, the assignments
+
+&ensp; &ensp; $s_1\gets 1$ , $f(2)\gets f(1)$
+
+are applied. 
+
+**btwxt** also enables *extrapolation* for estimation of the function outside the valid interpolation range. Two methods for extrapolation are available: *constant* and *linear*. If constant is selected, the function value for extrapolation is assumed to equal that of the nearest endpoint. A linear extrapolation returns the function value of the nearest endpoint at the grid-point coordinate for that endpoint, and has the slope assigned to that endpoint.
```

Comment:
Separate heading for extrapolation? This isn't specific to cubic interpolation. Maybe this should be moved after we're done discussing cubic interpolation?